src/OrleansTestKit/Storage/StorageExtensions.cs

```diff
@@ -25,5 +30,90 @@ public static TestStorageStats StorageStats(this TestKitSilo silo)
 
             return silo.StorageManager.GetStorageStats();
         }
+
+        public static IPersistentState<T> AddPersistentState<T>(
+            this TestKitSilo silo,
+            string stateName = default,
+            string storageName = default,
+            T state = default)
+            where T : new()
+        {
+            if (silo == null)
+            {
+                throw new ArgumentNullException(nameof(silo));
+            }
+
+            var storage = silo.StorageManager.GetStorage<T>(stateName);
+
+            return AddPersistentState(silo, storage, stateName, storageName, state);
+        }
+
+        public static IPersistentState<T> AddPersistentState<T>(
+            this TestKitSilo silo,
+            IStorage<T> storage,
+            string stateName = default,
+            string storageName = default,
+            T state = default)
+            where T : new()
+        {
+            if (silo == null)
+            {
+                throw new ArgumentNullException(nameof(silo));
+            }
+
+            silo.StorageManager.AddStorage(storage, stateName);
+            var stateMock = new PersistentStateFake<T>(storage)
+            {
+                State = state ?? new T()
+            };
+
+            var mockMapper = silo.StorageManager.stateAttributeFactoryMapperMock;
+
+            if (string.IsNullOrWhiteSpace(stateName))
```

Comment:
Is it possible to have a non-null `storageName` with a null `stateName`? Looks to be the only missing case?

Reply:
That's not possible based on the PersistantStateAttribute so i've updated the error handling for these cases.